app/uk/gov/hmrc/customs/inventorylinking/export/controllers/HeaderValidator.scala

```diff
@@ -44,79 +43,64 @@ class HeaderValidator @Inject()(logger: ExportsLogger) {
-      logger.error(s"Error - header '$headerName' value '$value' is not valid")
-      left
-    }
-
-    h.get(headerName).fold[Either[ErrorResponse, String]]{
-      leftWithLog(headerName)
-    }{
-      v =>
-        if (rule(v)) Right(v) else leftWithLogContainingValue(headerName, v)
-    }
-  }
-
-  def eitherBadgeIdentifier[A](allowNone: Boolean)(implicit vhr: HasRequest[A] with HasConversationId): Either[ErrorResponse, Option[BadgeIdentifier]] = {
-    val maybeBadgeId: Option[String] = vhr.request.headers.toSimpleMap.get(XBadgeIdentifierHeaderName)
-
-    if (allowNone && maybeBadgeId.isEmpty) {
-      logger.info(s"$XBadgeIdentifierHeaderName header empty and allowed")
-      Right(None)
-    } else {
-      maybeBadgeId.filter(xBadgeIdentifierRegex.findFirstIn(_).nonEmpty).map { b =>
-        logger.info(s"$XBadgeIdentifierHeaderName header passed validation: $b")
-        Some(BadgeIdentifier(b))
-      }.toRight[ErrorResponse] {
-        logger.error(s"$XBadgeIdentifierHeaderName invalid or not present for CSP")
-        errorResponseBadgeIdentifierHeaderMissing
-      }
+    h.get(headerName) match {
+      case Some(value) if rule(value) =>
+        Right(value)
+      case Some(invalidValue) =>
+        logger.error(s"Error - header '$headerName' value '$invalidValue' is not valid")
+        Left(errorResponse)
+      case None =>
+        logger.error(s"Error - header '$headerName' not present")
+        Left(errorResponse)
     }
   }
 
-  private def validEori(eori: String) = InvalidEoriHeaderRegex.findFirstIn(eori).isEmpty
+  def eitherBadgeIdentifier[A](implicit vhr: HasRequest[A] with HasConversationId): Either[ErrorResponse, Option[BadgeIdentifier]] = {
+    val maybeBadgeId = vhr.request.headers.toSimpleMap.get(XBadgeIdentifierHeaderName)
+    logger.debug(s"maybeBadgeId => $maybeBadgeId")
 
-  private def convertEmptyHeaderToNone(eori: Option[String]) = {
-    if (eori.isDefined && eori.get.trim.isEmpty) {
-      None
-    } else {
-      eori
+    maybeBadgeId match {
+      case Some(badgeId) if xBadgeIdentifierRegex.pattern.matcher(badgeId).matches =>
+        logger.info(s"$XBadgeIdentifierHeaderName header passed validation: $badgeId")
+        Right(Some(BadgeIdentifier(badgeId)))
+      case Some(_) =>
+        logger.error(s"$XBadgeIdentifierHeaderName invalid or not present for CSP")
+        Left(errorResponseBadgeIdentifierHeaderMissing)
+      case None =>
+        logger.info(s"$XBadgeIdentifierHeaderName header empty and allowed")
+        Right(None)
     }
   }
 
   def eoriMustBeValidIfPresent[A](implicit vhr: HasRequest[A] with HasConversationId): Either[ErrorResponse, Option[Eori]] = {
-    val maybeEoriHeader: Option[String] = vhr.request.headers.toSimpleMap.get(XSubmitterIdentifierHeaderName)
+    val maybeEoriHeader: Option[String] = vhr.request.headers.toSimpleMap.get(XSubmitterIdentifierHeaderName).filter(_.exists(!_.isWhitespace))
```

Comment:
It's a good point, a blank EORI should definitely fail validation so this bit of code should be redundant.

However, out of caution, I am preserving the original functionality (see the deleted method `convertEmptyHeaderToNone`) where an empty/blank EORI header should return `Right(None)` rather than a `Left`.

Reply:
Thank you for explaining. In that case wouldn't checking the string is not empty would do without checking if it exists? If the value doesn't exist, it won't go into filter. :)
val maybeEoriHeader: Option[String] = vhr.request.headers.toSimpleMap.get(XSubmitterIdentifierHeaderName).filter(!_.isBlank)

Reply:
I didn't use `isBlank` as it is only available in Java 11 and the project was compiling with JDK8 at the time of working on it.

But now that you mention it, https://github.com/hmrc/customs-inventory-linking-exports/pull/177 has been merged so  this should be available.

I will change as per your suggestion.

Reply:
Done.